internal/app/machined/internal/api/reg/reg.go

```diff
@@ -507,3 +507,38 @@ func getContainerInspector(ctx context.Context, namespace string, driver common.
 		return nil, fmt.Errorf("unsupported driver %q", driver)
 	}
 }
+
+// Read implements the read API.
+func (r *Registrator) Read(in *machineapi.ReadRequest, srv machineapi.Machine_ReadServer) (err error) {
+	stat, err := os.Stat(in.Path)
+	if err != nil {
+		return err
+	}
+
+	switch mode := stat.Mode(); {
+	case mode.IsRegular():
+		f, err := os.Open(in.Path)
+		if err != nil {
+			return err
+		}
+
+		ctx, cancel := context.WithCancel(context.Background())
+
+		defer cancel()
+
+		chunker := stream.NewChunker(f)
```

Comment:
that is not obvious, but `f` is going to be closed by `chunker`

Reply:
Hmm yeah, I'm not seeing _where_ though.

Reply:
should be https://github.com/talos-systems/talos/blob/master/pkg/chunker/stream/stream.go#L65

Reply:
Ahh I'm using the file chunker. That one doesn't seem to close the file.

Reply:
Should I use the file one? Been a while since I looked at these.